GBA/input.zig

```diff
@@ -1,78 +1,107 @@
-const GBA = @import("core.zig").GBA;
-
-pub const Input = struct {
-    var previousInput: u16 = 0;
-    var currentInput: u16 = 0;
-
-    pub const Keys = struct {
-        pub const A = 1 << 0;
-        pub const B = 1 << 1;
-        pub const Select = 1 << 2;
-        pub const Start = 1 << 3;
-        pub const Right = 1 << 4;
-        pub const Left = 1 << 5;
-        pub const Up = 1 << 6;
-        pub const Down = 1 << 7;
-        pub const R = 1 << 8;
-        pub const L = 1 << 9;
-    };
+//! Access to the key control and status registers should
+//! go through the functions in this module.
+//!
+//! Keys are active low, which doesn't play well intuitively
+//! with checking them. These functions abstract that conversion
+const std = @import("std");
+
+pub const Keys = std.EnumSet(Key);
+
+var prev_input: Keys = .{};
+var curr_input: Keys = .{};
+
+const REG_KEYINPUT: *align(2) const volatile u10 = @ptrFromInt(0x4000130);
```

Comment:
This should be in IO.zig no?

Reply:
I debated that, and I did have a pointer there at one point, but the active low combined with the current/previous input polling flow made me think it should maybe be encapsulated privately then maybe re-exposed through IO.zig. I guess it doesn't matter overly much which file the pointer is in if it isn't public and is only exposed through the public functions.

Reply:
Oh then it make sense to keep that register not visible outside the input module.